### PR TITLE
Integration with Java

### PR DESCRIPTION
**Pull request by jcnoriega**
### Summary of changes

A callback has been added to `Card.approveOrder` in order to facilitate integrations with merchants using Java. The callback pattern is the desired choice to use across the app

 ### Checklist

 - [ ] Added a changelog entry

- jonoriega
- sshropshire

### CODE REVIEW

**Comment by sarahkoop** [comment]
This looks like a good approach to me! Do we have any concerns about it?

**Comment by sarahkoop** [comment]
This looks like a good approach to me! Do we have any concerns about it?

**Comment by sshropshire** [comment]
@sarahkoop there is a bit of a choice paradox if we keep them in the same module. Auto complete in Java still shows the coroutine method. There is a chance that some merchants may try to invoke the coroutine version and that'll send them down a rabbit hole.

**Comment by sarahkoop** [comment]
It seems like the Jetpack libraries are Java support first and then have `-ktx` extension libraries. We could go that route so that only merchants who include the `ktx` library get the coroutine functionality.  But that would mean our core SDK is using the more legacy features.  This could be the best future-proof path though.

I think we could have auto-complete include both methods and rely on merchants reading the documentation before using a method. I suppose we might get more inbounds, but hopefully they would be quick to redirect to the docs and close.

File: Card/build.gradle [hunk 1]
```diff
@@ -55,6 +55,9 @@ dependencies {
 
     androidTestImplementation deps.androidxJUnit
     androidTestImplementation deps.androidxEspressoCore
+
+    implementation 'org.jetbrains.kotlinx:kotlinx-coroutines-jdk8:1.3.9'
```
Comment (sarahkoop) [inline Card/build.gradle]: Nitpick but let's put this with the rest of the `deps`

File: Card/src/main/java/com/paypal/android/card/CardClient.kt [hunk 2]
```diff
@@ -18,4 +22,26 @@ class CardClient internal constructor(private val cardAPI: CardAPI) {
      */
     suspend fun approveOrder(request: CardRequest): CardResult =
```
Comment (sarahkoop) [inline Card/src/main/java/com/paypal/android/card/CardClient.kt:23]: Should we call out in this method docs to only use with Kotlin integration?

File: Card/src/main/java/com/paypal/android/card/CardClient.kt [hunk 3]
```diff
@@ -18,4 +22,26 @@ class CardClient internal constructor(private val cardAPI: CardAPI) {
...
+                val result = approveOrder(request)
+                callback.success(result)
+            } catch (e: PayPalSDKError) {
+                callback.failure(e)
+            }
+        }
+    }
+}
+
+interface ApproveOrderCallback {
```
Comment (sarahkoop) [inline Card/src/main/java/com/paypal/android/card/CardClient.kt]: Maybe this should live in its own file and have doc strings? Not sure what convention we want to follow for callbacks in this project.

**Review by sarahkoop — APPROVED** [review]
A few comments, but looks good!

**Review by andrespelaezp — APPROVED** [review]
Looking good!

**Pull request merged**